edb/edgeql-parser/src/tokenizer.rs

```diff
@@ -296,6 +296,11 @@ impl<'a> TokenStream<'a> {
                                 "backtick-quoted name cannot \
                                     start with char `@`"));
                         }
+                        if val.starts_with("`$") {
+                            return Err(Error::unexpected_static_message(
+                                "backtick-quoted name cannot \
```

Comment:
I think "backtick-quoted" is redundant in the message.  These characters are banned in all identifiers, quoted or not.

Reply:
True, but only in a very technical sense.
```
SELECT <str>$ident;   # is legal
SELECT <str>`$ident`; # is not legal
```
And to most people the subtlety of `$ident` being a "dollar, followed by an identifier" vs being a special identifier is obscure, especially since parameters work very much like identifiers.

Whereas the error message gets to the core of it, that you cannot write certain things inside backticks.